src/ConnectedMode/ServerSentEvents/ServerSentEventsFilter.cs

```diff
@@ -0,0 +1,29 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
+{
+    internal interface IServerSentEventsFilter
+    {
+        IServerEvent GetFilteredEventOrNull(IServerEvent serverEvent);
```

Comment:
What is the purpose of this interface? it receives the event, so does it do somewhere a switch-case/if-else? if so, this is a repetition, the split should be done in a single place, before where send to the corresponding publisher.

Reply:
I mainly introduced this to get unblocked because I didn't have the models to actually implement the filtering

Reply:
I will have to implement this in a separate PR right after this one.

Reply:
Might be better to just remove it now, since this is a new interface and we know it won't be used. So no point in committing code that won't be used.

Reply:
It will break the tests

Reply:
Then fix them :P

Reply:
To elaborate: otherwise I'm reviewing code that is irrelevant.